### PR TITLE
TST: misplaced reduction/indexing tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -1274,3 +1274,28 @@ def test_series_broadcasting(self):
             df_nan.clip(lower=s, axis=0)
             for op in ["lt", "le", "gt", "ge", "eq", "ne"]:
                 getattr(df, op)(s_nan, axis=0)
+
+
+class TestDataFrameReductions:
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_analytics.py:1279]: There is also a `tests/reductions/` directory with a lot of min/max tests. What is the general idea here of what kind of reductions tests should be placed where?

Comment (jbrockmendel) [reply pandas/tests/frame/test_analytics.py:1279]: tests.reduction is analogous to tests.arithmetic, intended for tests that are shared across classes

Comment (jreback) [reply pandas/tests/frame/test_analytics.py:1279]: hmm, you put the series ones in test_reductions.py I think these should be in a similar one?

Comment (jbrockmendel) [reply pandas/tests/frame/test_analytics.py:1279]: yah considered that, but for frame test_analytics contains most of the reduction tests.  i think the thing to do is get non-reduction tests out of there and then rename to test_reduction

Comment (jreback) [reply pandas/tests/frame/test_analytics.py:1279]: kk fair, keep in mind

**Pull request merged**